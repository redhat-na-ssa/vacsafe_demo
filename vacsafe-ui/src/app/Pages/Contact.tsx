import React from 'react';
import { Button, Card, CardBody, CardFooter } from '@patternfly/react-core';
import { ExternalLinkAltIcon } from '@patternfly/react-icons';
import CardWithTitle from '@app/Components/CardWithTitle';
import { useHistory } from 'react-router-dom';

const Contact: React.FunctionComponent = () => {
  const history = useHistory();
  return (
    <Card isRounded>
      <CardWithTitle
        title="Help"
        info="If you need more information, use link below."
      />
      <CardBody>
        <Button
          icon={<ExternalLinkAltIcon />}
          variant="link"
          component="a"
          href="https://oshr.nc.gov/covid-19-vaccination-or-testing-faqs"
          target="_blank"
        >
          Vaccination and Testing FAQ
        </Button>
      </CardBody>
      <CardFooter>
        <Button variant="plain" onClick={() => history.push("/")}>
          Back
        </Button>
      </CardFooter>
    </Card>
  );
};

export { Contact };
