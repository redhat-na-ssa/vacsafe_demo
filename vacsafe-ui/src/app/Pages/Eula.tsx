import React from 'react';
import {
  Button, 
  Card,
  CardFooter,
  CardBody,
  Stack,
  Text,
} from '@patternfly/react-core';
import { CheckCircleIcon } from '@patternfly/react-icons'
import BrowserSupportWarning from '@app/Components/BrowserSupportWarning';
import CardWithTitle from '@app/Components/CardWithTitle';
import { useHistory } from 'react-router-dom';
import { useKeycloak } from "@react-keycloak/web";

const Eula: React.FunctionComponent = () => {
  const history = useHistory();
  const { keycloak } = useKeycloak();

  function login() {
    if(!keycloak?.authenticated) {
      keycloak?.login()
    }
  }
  if(keycloak?.authenticated) {
    sessionStorage.setItem("token", keycloak.token!);
    keycloak.loadUserProfile().then(() => {
      history.push("/profile")
    })
  }

  return (     
    <Card isRounded>
      <CardWithTitle title="Welcome" info="End User Licensing Agreement" />
      <CardBody>
        <Stack hasGutter>
          <Text>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi metus lacus, aliquet a molestie vel, pharetra ac turpis. 
          Etiam ac tellus in mauris scelerisque sollicitudin cursus quis diam. Nullam malesuada sit amet justo eu tristique. 
          Suspendisse accumsan volutpat augue vitae mollis. Proin lacinia arcu eu ligula malesuada, convallis sodales urna convallis. 
          Aliquam erat volutpat. Cras et neque aliquam, rhoncus nunc at, eleifend ligula. Nunc mattis dui nec justo facilisis euismod.
          Mauris ultrices vehicula ligula, et convallis mi rutrum et. Aenean quis tincidunt dui, convallis volutpat magna. Aliquam 
          in odio non augue venenatis placerat ut et urna. Vivamus elementum pretium sagittis. Pellentesque habitant morbi tristique
          senectus et netus et malesuada fames ac turpis egestas. Suspendisse suscipit nibh pretium posuere tincidunt. Etiam rhoncus
          magna odio, sed tempus libero posuere et. Curabitur viverra tellus in sodales feugiat.</Text>
          <Text>
          Duis id euismod augue. In eu eros odio. Praesent lacinia pharetra quam, sit amet fringilla justo accumsan ut. Suspendisse 
          pharetra leo at iaculis dictum. Nullam sed nulla quis dui varius bibendum aliquam id diam. Lorem ipsum dolor sit amet, 
          consectetur adipiscing elit. Phasellus condimentum vel magna in gravida. Sed laoreet ante vel imperdiet sagittis. Cras 
          luctus, metus vitae pretium condimentum, turpis turpis porta lacus, ac ultrices ex nibh eu nunc.</Text>
          <BrowserSupportWarning />
        </Stack>
      </CardBody>
      <CardFooter>
        <Button
          icon={<CheckCircleIcon />}
          variant="primary"
          onClick={() =>( login() )}>
          Accept
        </Button>
      </CardFooter>
    </Card>  
  );
};

export { Eula };
