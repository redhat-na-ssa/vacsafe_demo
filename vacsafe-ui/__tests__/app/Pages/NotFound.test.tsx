import React from 'react';
import { shallow } from 'enzyme';
import { NotFound } from '../../../src/app/Pages/NotFound';

describe('NotFound Page', ()=> {
    it('should be defined', () => {
        expect(NotFound).toBeDefined()
    })
    it("should render without crashing", () => {
        const wrapper = shallow(<NotFound />)
        expect(wrapper)
    })
    it("should match existing snapshot", () => {
        const wrapper = shallow(<NotFound />)
        expect(wrapper).toMatchSnapshot()
    })
})
