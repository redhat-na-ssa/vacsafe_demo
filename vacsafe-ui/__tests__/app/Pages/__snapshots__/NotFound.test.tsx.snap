// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Page should match existing snapshot 1`] = `
<PageSection>
  <EmptyState
    variant="full"
  >
    <EmptyStateIcon
      icon={[Function]}
    />
    <Title
      headingLevel="h1"
      size="lg"
    >
      404 Page not found
    </Title>
    <EmptyStateBody>
      We didn't find a page that matches the address you navigated to.
    </EmptyStateBody>
    <GoHomeBtn />
  </EmptyState>
</PageSection>
`;
